certbot/account.py

```diff
@@ -117,6 +117,18 @@ def load(self, account_id):
         except KeyError:
             raise errors.AccountNotFound(account_id)
 
+class RegistrationResourceWithNewAuthzrURI(messages.RegistrationResource):
+    """A backwards-compatible RegistrationResource with a new-authz URI.
+
+       Hack: Certbot versions pre-0.11.1 expect to load
+       new_authzr_uri as part of the account. Because people
+       sometimes switch between old and new versions, we will
+       continue to write out this field for some time so older
+       clients don't crash in that scenario. Because this is a
+       temporary hack, just hardcode the known new-authz URI for
+       Let's Encrypt.
+    """
+    new_authzr_uri = jose.Field('new_authzr_uri')
```

Comment:
Old clients will indeed use the saved value. I was being lazy here because I think the URL is very unlikely to change during the lifetime of the v1 API. But I've pushed some changes to do the more rigorous thing. If you think they look good, I'll rebase this on top of a revert of the revert.